### ISSUE TITLE
Layer Re-ordering is not working within Web component.

### ISSUE THREAD

**Issue by venkatesh-brightly** [Viglino/ol-ext#1048]
I have a web component **'my-map'** that is rendering openlayer map. I am using LayerShop Class for layer switcher. 

**Problem:**
When I am trying to re-order the layers, I am not able to drop the layer on top or bottom.

_It's working fine outside web component. **But not inside web component**._

**Comment by Viglino** [comment]
Do you have an example online to test.
What sort of component are you using ?

**Comment by venkatesh-brightly** [comment]
Please check this link: https://codesandbox.io/p/sandbox/pedantic-leaf-hf2z7d?file=%2Fsrc%2Findex.js

While debugging, I figured out that in **move(e)** function, **e.target** and **document.elementFromPoint** is actually returning the web component instead of the actual node which is preventing it from dropping.

I fixed it by updating the below code for now:
```
var li
if (!e.touches) {
  li = e.target
  /**
   *  Get the HTML node within web component on click drag
   */
  if(e.target.shadowRoot){
    li = e.composedPath()[0]
  }
} else {
  li = document.elementFromPoint(e.touches[0].clientX, e.touches[0].clientY);
  /**
   *  Get actual HTML node within web component on touch drag
   */
  while(li.shadowRoot){
    li = li.shadowRoot.elementFromPoint(e.touches[0].clientX, e.touches[0].clientY)
  }
}
```

This code can be found in file: **LayerSwitcher.js:594**

**Comment by venkatesh-brightly** [comment]
When will these changes published on npm?

**Comment by Viglino** [comment]
Just published [ol-ext@4.0.18](https://www.npmjs.com/package/ol-ext)
